EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -135,63 +146,150 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
             //mlContext.Model.Save(trainedModel, trainingData.Schema, @"C:\Users\User\Downloads\TrainedModel.zip");
         }
 
-        private static Dictionary<string, int> GetSequenceToPSMCount(List<PeptideSpectralMatch> psms)
+        /// <summary>
+        /// Assuming that we want to use different sets of training features for different search types, the search type needs to be determined.
+        /// In the future, we may have that specified in the GUI and this will be moot.
+        /// </summary>
+        /// <param name="psms"></param>
+        /// <returns></returns>
+        private static string DetermineSearchType(List<PeptideSpectralMatch> psms)
```

Comment:
I like the design here. This can help extend the function for crosslink search or others...